frontend/src/contexts/SearchContext.tsx

```diff
@@ -19,6 +20,18 @@ const SearchProvider = ({ children, initialQuery }: SearchProviderProps) => {
     }
   }, [q])
 
+  events.on('routeChangeComplete', (url: string) => {
+    /*
+     * Handle history navigation by update the search query in the context when the url changes.
+     * We read the query parameter from the url instead of using the router query
+     * because the router query gets updated asynchronously.
+     */
+    const latestQuery = new URLSearchParams(url.split('?')[1]).get('q')
+    if (latestQuery !== null) {
```

Comment:
```suggestion
    if (latestQuery != null) {
```

If there is no "?" in the `url` then latestQuery will be `undefined` which `!== null`. Generally when checking for null/undefined I just use the lax equality check instead of the stricter triple one.